impl/httpMetricsManager.go

```diff
@@ -67,20 +66,31 @@ func (m *HttpMetricsManager) GetThroughput() int {
 
 // return the sampletime(duration) at input percentile
 func (m *HttpMetricsManager) GetDurationAt(thre int) time.Duration {
-	if !sort.IsSorted(m.records) {
-		sort.Sort(m.records)
+	targetCount := thre * m.numRecords / 100
```

Comment:
This function should probably take in all of the requested thresholds. One threshold at a time, means the sorting and looping below will have to be done for every threshold.